portfolio/src/main/webapp/UserHandler.js

```diff
@@ -0,0 +1,42 @@
+/** The class which checks user's login status. */
+export default class UserHandler {
+  
+  /** Gets user's login status and helpful information.
+   * If user has logged in, show the commentbox and username;
+   * if not, show the log in url.
+   * @param {Controller} controller The Controller of current webapp
+   */
```

Comment:
add `@private` to have the compiler make sure this is not used outside the instance

Reply:
done!